From 3e39d7abde526d9058527bb3f25358c991ac47e2 Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Tue, 18 Apr 2023 11:34:54 +0100
Subject: [PATCH 724/726] configs: Enable LRU_GEN_ENABLED on non-pi4 32-bit
 kernels

We believe the OOM issue with MGLRU on 32-bit kernel
only affects Pi4 which uses CONFIG_VMSPLIT_3G.

Pi0-3 which use CONFIG_VMSPLIT_2G is expected not to
suffer from this OOM issue.

So lets enable MGLRU by default for these users,
who many benefit from the advantages it has.

See: https://github.com/raspberrypi/linux/issues/5395
Signed-off-by: Dom Cobley <popcornmix@gmail.com>
---
 arch/arm/configs/bcm2709_defconfig | 1 +
 arch/arm/configs/bcmrpi_defconfig  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/arm/configs/bcm2709_defconfig b/arch/arm/configs/bcm2709_defconfig
index 6b7e7a7f522d..ef4e2bce0dca 100644
--- a/arch/arm/configs/bcm2709_defconfig
+++ b/arch/arm/configs/bcm2709_defconfig
@@ -69,6 +69,7 @@ CONFIG_Z3FOLD=m
 # CONFIG_COMPAT_BRK is not set
 CONFIG_CMA=y
 CONFIG_LRU_GEN=y
+CONFIG_LRU_GEN_ENABLED=y
 CONFIG_NET=y
 CONFIG_PACKET=y
 CONFIG_UNIX=y
diff --git a/arch/arm/configs/bcmrpi_defconfig b/arch/arm/configs/bcmrpi_defconfig
index 997e8a65f2b6..afef35a26897 100644
--- a/arch/arm/configs/bcmrpi_defconfig
+++ b/arch/arm/configs/bcmrpi_defconfig
@@ -64,6 +64,7 @@ CONFIG_Z3FOLD=m
 # CONFIG_COMPAT_BRK is not set
 CONFIG_CMA=y
 CONFIG_LRU_GEN=y
+CONFIG_LRU_GEN_ENABLED=y
 CONFIG_NET=y
 CONFIG_PACKET=y
 CONFIG_UNIX=y
-- 
2.33.1

